### DISCUSSION TITLE
Execute DynamicOutputs in serial order instead of in parallel

### DISCUSSION THREAD

**Discussion by lbstoller** [dagster-io/dagster#12016] [Q&A] [answered]
Hi. I might have missed how to accomplish this in the documentation ... but say I return a list of dynamic outputs and apply that list to a @graph() with multiple ops. When I do that, they all run in parallel. But I need them to run serially. Is there a way to do that?

Thanks

**Comment by alangenfeld** [comment, marked as answer]
You can configure the executor to enforce concurrency limits

https://docs.dagster.io/concepts/ops-jobs-graphs/job-execution#op-concurrency-limits

**Comment by lbstoller** [comment]
Hi. Like this? I did this and it had no effect, they still ran in parallel. 

```
        "execution": {
            "config": {
                "multiprocess": {
                    "max_concurrent": 1
                }
            }
        }
```

**Reply by alangenfeld** [reply]
I'm surprised that had no effect - I would expect all ops to execute in serial with that configuration. 

You could also use `tag_concurrency_limits` in conjunction with a tag on the specific op you need to run in serial to enforce concurrency only on that op.

**Comment by lbstoller** [comment]
Ah, I think I see the confusion. Consider this fragment. What happens is that a bunch of step1 ops run. Then a bunch of step2. Then a bunch of step3. That all happens serially. But I need the entire *graph* to run (step1,step2,step3), followed by the next whole graph (step1,step2,step3) for all of the dynamic outputs in the list. 

```
@op()
def step1(val):
    return val

@op()
def step2(val):
    return val

@op()
def step3(val):
    return val

@graph()
def run(val):
    return step3(step2(step1(val)))

@job()
def job():
    outputs = return_dynamic()
    outputs.map(run)
    pass
```

**Reply by alangenfeld** [reply]
https://github.com/dagster-io/dagster/discussions/8871

**Comment by lbstoller** [comment]
Perfect, that works great! Thank you very much.

**Comment by rexledesma** [comment]
Just to close this thread, the answer is a duplicate of https://github.com/dagster-io/dagster/discussions/8871.
